lib/common/src/Dom/Document.php

```diff
@@ -1446,6 +1458,16 @@ public function isValidHeadNode(DOMNode $node)
         );
     }
 
+    /**
+     * Determine whether `data-ampdevmode` was initially set on the document element.
+     *
+     * @return bool
+     */
+    public function hasInitialAmpDevMode()
```

Comment:
This method should be tested in `DocumentTest`, verifying the method returns:

1. `false` when the `data-ampdevmode` attribute is absent on `html`.
1. `false` when the `data-ampdevmode` attribute is added on `html` after parsing.
1. `true`  when the `data-ampdevmode` attribute is initially on `html`.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4440/commits/3081a0a5de6e45c3d01ea0c9059f9914dc37128b.